modified Wi-SUN tasklet to return success when calling connect twice

mikter opened the pull request:
### Description

If connect to be called multiple times it will return NSAPI_ERROR_IS_CONNECTED status.

### Pull request type

    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [X] Breaking change

### Reviewers

@JarkkoPaso @artokin @SeppoTakalo @AnttiKauppila @teetak01 

### Release Notes

##### Summary of changes
As stated in Mesh API documentation, we should return the status NSAPI_ERROR_IS_CONNECTED if connect is called twice. This PR (11556) fixes it.

##### Impact of changes

If connect method is called twice then different status code is returned.

##### Migration actions required

Implementations that rely on NSAPI_ERROR_DEVICE_ERROR status code when calling connect twice as their recovery method needs to change the handling to the correct status NSAPI_ERROR_IS_CONNECTED.

artokin commented on features/nanostack/mbed-mesh-api/source/LoWPANNDInterface.cpp:
```diff
@@ -171,7 +171,7 @@ bool LoWPANNDInterface::getRouterIpAddress(char *address, int8_t len)
 }
 
 #define LOWPAN 0x2345
-#if MBED_CONF_NSAPI_DEFAULT_MESH_TYPE == LOWPAN && DEVICE_802_15_4_PHY
```
Why 802_15_4_PHY is not requires for mesh anymore?

artokin commented on features/nanostack/mbed-mesh-api/source/wisun_tasklet.c:
```diff
@@ -133,6 +133,7 @@ static void wisun_tasklet_main(arm_event_s *event)
              * The event is delivered when the NanoStack OS is running fine.
              * This event should be delivered ONLY ONCE.
              */
+            tr_debug("event_Init");
```
too detailed trace

artokin commented on features/nanostack/mbed-mesh-api/source/wisun_tasklet.c:
```diff
@@ -434,7 +436,7 @@ int8_t wisun_tasklet_connect(mesh_interface_cb callback, int8_t nwk_interface_id
 
     if (wisun_tasklet_data_ptr->network_interface_id != INVALID_INTERFACE_ID) {
-        return -3;  // already connected to network
+        return wisun_tasklet_data_ptr->tasklet;  // already connected to network
```
Is this change needed also to other mesh tasklets?

0xc0170 commented on features/netsocket/mbed_lib.json:
```diff
@@ -40,7 +40,7 @@
         "default-mesh-type": {
             "help": "Configuration type for MeshInterface::get_default_instance(). [LOWPAN/THREAD/WISUN]",
-            "value": "THREAD"
+            "value": null
```
one question before integrating this: isn't this breaking change for apps - as the config value changes for default mesh type ?

0xc0170 replied:
As discussed, these changes are potentially breaking, adding proper labels

SeppoTakalo replied:
Questionable..
Is it a breaking change to return no default, instead of non-configured non-working default?

Or @mikter did the non-configured Thread work? My assumption is that it is a demo, with no real use cases.. should we optimise for testing and demo, or for real use cases.

mikter replied:
I don't think anyone should have Thread configured without any configuration in application 

It would however work as a non-commissioned Thread device, but that should still be application configuration not a default. 

I can remove this change if you like from this commit as it is not really related. although the Breaking change is not only because of this configuration, but it is because we are changing status code on public API.

0xc0170 reviewed (changes requested):
This needs "Release notes section"

0xc0170 reviewed (commented):
It looks fine but commit message should contain more details for the change like this (copy paste what you included in the description at least, or text from the release notes).

Just amend the last commit message and we shall proceed

Commit:
Modified Wi-SUN tasklet to return correct status when calling connect twice

When calling mesh connect method twice NSAPI_ERROR_IS_CONNECTED is returned
instead of NSAPI_ERROR_DEVICE_ERROR Following the mesh API documentation.

juhhei01 commented on features/nanostack/nanostack-interface/Nanostack.cpp:
```diff
@@ -116,7 +116,7 @@ nsapi_error_t map_mesh_error(mesh_error_t err)
         case MESH_ERROR_PARAM:
             return NSAPI_ERROR_PARAMETER;
         case MESH_ERROR_STATE:
-            return NSAPI_ERROR_DEVICE_ERROR;
```
Looks like that now we are hiding something bigger issue? If we can return connected state at mesh error state.

juhhei01 replied:
Current return value make more sense if we are already connected
